llvm/lib/CodeGen/RegisterCoalescer.cpp

```diff
@@ -3599,6 +3634,10 @@ bool RegisterCoalescer::joinVirtRegs(CoalescerPair &CP) {
   if (!LHSVals.resolveConflicts(RHSVals) || !RHSVals.resolveConflicts(LHSVals))
     return false;
 
+  // Some CR_Replace can't be solved by pruneValues. Early exit here.
```

Comment:
Instead of fixing this up after the fact, you should avoid reporting CR_Replace in the first place